src/Connector.SqlServer/Connector/SqlServerConnector.cs

```diff
@@ -315,9 +315,23 @@ public override async Task<ConnectionVerificationResult> VerifyConnection(Execut
 
                 await using var connectionAndTransaction = await _client.BeginTransaction(configurationData);
                 var connectionIsOpen = connectionAndTransaction.Connection.State == ConnectionState.Open;
+
+                if (!connectionIsOpen)
+                {
+                    return new ConnectionVerificationResult(false);
+                }
+
+                var schema = configurationData.GetValue(SqlServerConstants.KeyName.Schema, (string)null);
+                if (string.IsNullOrEmpty(schema))
+                {
+                    schema = SqlTableName.DefaultSchema;
+                }
+
+                var schemaExists = await _client.VerifySchemaExists(connectionAndTransaction.Transaction, schema);
+
```

Comment:
It is going to be hard to distinguish if verification false is due to connectivity or permissions to the schema.
Should we log something here?